### PR TITLE
VR-1672: Enable Optional DataVer Imports

### PR DESCRIPTION
**Pull request by convoliution**
## Changelog
- cleanly handle `pillow`, `google.cloud.bigquery`, and `boto3` optional imports
- remove dependency on `pytz`

### CODE REVIEW

File: verta/verta/client.py [hunk 1]
```diff
@@ -862,8 +862,7 @@ def get_s3_object_info(object_info, key=None):
         dataset_part_info.path = object_info['Key'] if key is None else key
         dataset_part_info.size = object_info['Size'] if key is None else object_info['ContentLength']
         dataset_part_info.checksum = object_info['ETag']
-        dataset_part_info.last_modified_at_source = int((object_info['LastModified']
-                                                       - datetime.datetime(1970, 1, 1, tzinfo=pytz.UTC)).total_seconds())
+        dataset_part_info.last_modified_at_source = int(object_info['LastModified'].timestamp())
```
Comment (convoliution) [inline verta/verta/client.py]: `object_info['LastModified']` is already timezone-aware.

File: verta/verta/client.py [hunk 2]
```diff
@@ -971,7 +970,7 @@ def __init__(self, job_id=None,
         if job_id is not None and location:
             self.job_id = job_id
             job = self.get_bq_job(job_id, location)
-            self.execution_timestamp = int((job.started - datetime.datetime(1970,1,1, tzinfo=pytz.UTC)).total_seconds())
+            self.execution_timestamp = int(job.started.timestamp())
```
Comment (convoliution) [inline verta/verta/client.py]: `job.started` is already timezone-aware

File: verta/verta/client.py [hunk 3]
```diff
@@ -2998,7 +3000,8 @@ def get_image(self, key):
         else:
             try:
                 return PIL.Image.open(six.BytesIO(image))
-            except IOError:
+            except (NameError,  # Pillow not installed
```
[the remark this answers is not in the document]
Comment (convoliution) [reply verta/verta/client.py]: We _do_ do "just fine without it".

But it's a convenience for `get_image()` to return objects that can render directly into images in a Jupyter Notebook. One of our demo Notebooks uses this, and one of our customers had used this early on (maybe just for exploration, but it was used nonetheless).

I do not intend to completely remove it.

File: verta/verta/client.py [hunk 4]
```diff
@@ -844,8 +841,11 @@ def __init__(self, bucket_name, key=None, url_stub=None):
         self.compute_dataset_size()
 
     def get_dataset_part_infos(self):
+        try:
+            conn = BotoClient('s3')
+        except NameError:  # Boto 3 not installed
+            six.raise_from(ImportError("Boto 3 is not installed"), None)
```
[the remark this answers is not in the document]
Comment (convoliution) [reply verta/verta/client.py]: Will do!

File: verta/verta/client.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (conradoverta) [reply verta/verta/client.py]: Ah, ok. So using PIL enables showing in the notebook. Cool.

**Comment by convoliution** [comment]
@conradoverta 

> I'd generally the recommend setting a variable with the same name as the module as None. Using name error can hide a lot of different other issues and isn't specific (e.g. typos in variable names will be classified as the error). We should be as specific as we can to reduce false positives.

I'm concerned that setting, say, `bigquery = None` introduces even more problems. For example, this means that, say, `bigquery.Client(params.url)` would raise an `AttributeError` instead of a `NameError`, which also has false positives (e.g. the identifier `Client` has a typo, or `params` has a type we didn't expect).

Essentially, my assumption is that the suggestion changes the error to catch from `NameError` to `AttributeError`. `AttributeError` can be caused by all sorts of runtime conditions, but `NameError` can only be caused by a write-time condition (a typo). So to me, `NameError` is safer.

**Pull request merged**